haas/client/base.py

```diff
@@ -0,0 +1,50 @@
+""" This module implements the HaaS client library. """
+
+import requests
+import os
+import json
+from haas.client import errors
+from urlparse import urljoin
+
+
+class ClientBase(object):
+    """ Main class which contains all the methods to
+    -- ensure input complies to API requisites
+    -- generates correct format for server API on behalf of the client
+    -- parses output from received from the server.
+    In case of errors recieved from server, it will generate appropriate
+    appropriate message.
+    """
+
+    def __init__(self, endpoint=None, sess=None):
```

Comment:
I did not understand your comment about `if these are mandatory just don't supply the defaults.`
I am checking if they are explicitly `None` and raises error in that case.